File: backend/app/models/address.py
from sqlalchemy import Column, String, Float
from app.database import Base
from app.models.base import TimestampMixin, UUID_TYPE, gen_uuid


class Address(Base, TimestampMixin):
    __tablename__ = "addresses"

    id = Column(UUID_TYPE, primary_key=True, default=gen_uuid)
    label = Column(String(50), nullable=True)          # "Home", "Office" etc
    full_address = Column(String(500), nullable=False)
    city = Column(String(100), nullable=True)
    contact_name = Column(String(150), nullable=True)
    contact_phone = Column(String(20), nullable=True)

    # Server-side geocoded, never trust client-supplied coordinates for pricing
    lat = Column(Float, nullable=True)
    lng = Column(Float, nullable=True)

File: mobile/lib/models/address.dart
class Address {
  final String? label;
  final String fullAddress;
  final String? city;
  final String? contactName;
  final String? contactPhone;
  final double? lat;
  final double? lng;

  Address({
    this.label,
    required this.fullAddress,
    this.city,
    this.contactName,
    this.contactPhone,
    this.lat,
    this.lng,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      label: json['label'] as String?,
      fullAddress: json['full_address'] as String? ?? '',
      city: json['city'] as String?,
      contactName: json['contact_name'] as String?,
      contactPhone: json['contact_phone'] as String?,
      lat: (json['lat'] as num?)?.toDouble(),
      lng: (json['lng'] as num?)?.toDouble(),
    );
  }
}
